### PR TITLE
[WF-342, WF-343] Serialize integrations and set PostgreSQL testing profile for just test

### LINKED ISSUE #18: Race condition due to missing `depends_on` for integrations
### Bug Description

There is a race condition that happens when Terraform tries to create integrations before charms are fully ready. The Juju provider does not always wait for apps to be available before attempting integrations AFAICT.

A way to workaround this would be to re-apply the module, but since we want to guarantee reproducibility, we need to add a solution directly in the module.

### Proposed solution

Explicitly add `depends_on` to each integration, referencing the module it should be integrated with, for example:

```
# Temporal Frontend ↔ PostgreSQL
resource "juju_integration" "frontend_to_postgresql" {
  model_uuid = var.model_uuid
  application {
    name     = "temporal-frontend"
    endpoint = module.temporal_frontend.requires.db
  }
  application {
    name     = var.postgresql.app_name
    endpoint = module.postgresql.provides.database
  }
  depends_on = [module.temporal_frontend, module.postgresql]
}
```

### To Reproduce

1. (with some tweaks) `just apply`
2. Check the status of the applications
3. Wait for the terraform apply command to error out

### Environment

* Canonical k8s v1.32.11
* Juju 3.6.19-genericlinux-amd64
* Terraform (snap) 1.14.7

### Relevant log output

```shell
19-bed1f36e94ba:postgres:database:temporal-worker:visibility]                                           
module.postgresql.juju_application.k8s_postgresql: Creation complete after 1m5s                         
[id=ee734531-1459-4e9f-8819-bed1f36e94ba:postgres]                                                      
╷                                                                                                       
│ Error: Client Error                                                                                   
│                                                                                                       
│   with juju_integration.frontend_to_postgresql,                                                       
│   on integrations.tf line 2, in resource "juju_integration" "frontend_to_postgresql":                 
│    2: resource "juju_integration" "frontend_to_postgresql" {                                          
│                                                                                                       
│ Unable to create integration, got error: the applications were not available to be integrated         
╵                                                                                                       
╷                                                                                                       
│ Error: Client Error                                                                                   
│                                                                                                       
│   with juju_integration.frontend_visibility_to_postgresql,                                            
│   on integrations.tf line 15, in resource "juju_integration" "frontend_visibility_to_postgresql":     
│   15: resource "juju_integration" "frontend_visibility_to_postgresql" {                               
│                                                                                                       
│ Unable to create integration, got error: the applications were not available to be integrated         
╵                                                                                                       
╷                                                                                                       
│ Error: Client Error                                                                                   
│                                                                                                       
│   with juju_integration.frontend_to_ui,                                                               
│   on integrations.tf line 28, in resource "juju_integration" "frontend_to_ui":                        
│   28: resource "juju_integration" "frontend_to_ui" {                                                  
│                                                                                                       
│ Unable to create integration, got error: the applications were not available to be integrated         
╵                                                                                                       
╷                                                                                                       
│ Error: Client Error                                                                                   
│                                                                                                       
│   with juju_integration.matching_visibility_to_postgresql,                                            
│   on integrations.tf line 92, in resource "juju_integration" "matching_visibility_to_postgresql":     
│   92: resource "juju_integration" "matching_visibility_to_postgresql" {                               
│                                                                                                       
│ Unable to create integration, got error: the applications were not available to be integrated         
╵                                                                                                       
╷                                                                                                       
│ Error: Client Error                                                                                   
│                                                                                                       
│   with juju_integration.admin_to_matching,                                                            
│   on integrations.tf line 118, in resource "juju_integration" "admin_to_matching":                    
│  118: resource "juju_integration" "admin_to_matching" {                                               
│                                                                                                       
│ Unable to create integration, got error: the applications were not available to be integrated         
╵                                                                                                       
error: Recipe `apply` failed with exit code 1
```

### Additional context

_No response_

### LINKED ISSUE #19: Exceeding PostgreSQL connections in CI testing
### Bug Description

We are currently exhausting PostgreSQL's `max_connections`, likely because of the combination of the various Temporal applications all establishing connections simultaneously during setup.

### Proposed solution

For testing purposes, we can try increasing the `max_connections` by setting the `profile=testing` configuration option. This will guarantee this runs on the CI w/o issues.

### To Reproduce

1. (with some tweaks) `just apply` the module
2. Observe any of the Temporal applications going into `error` state (it can be any because the error can be caused by any connection not being established "on time" before the connections are exceeded). For example:

```
Model          Controller  Cloud/Region  Version  SLA          Timestamp                                 
temporal-test  ck8s        k8s           3.6.19   unsupported  16:28:42-06:00                            
                                                                                                         
App                Version  Status  Scale  Charm               Channel    Rev  Address         Exposed  M
essage                                                                                                   
postgres           14.20    active      1  postgresql-k8s      14/stable  774  10.152.183.215  no        
temporal-admin     1.23.1   error       1  temporal-admin-k8s  1.23/edge   25  10.152.183.174  no       h
ook failed: "admin-relation-changed"       
```
3. Check `juju debug-log` for details

### Environment

- Canonical k8s v1.32.11
- Juju 3.6.19-genericlinux-amd64
- Terraform (snap) 1.14.7

### Relevant log output

```shell
22:25:51.705Z\t\x1b[31mERROR\x1b[0m\tUnable to connect to SQL database.\t{"error": "pq: remaining connect
ion slots are reserved for non-replication superuser connections", "logging-call-at": "handler.go:52"}\n'
unit-temporal-admin-0: 16:25:51 ERROR unit.temporal-admin/0.juju-log admin:21: Uncaught exception while i
n charm code:                                                                                            
Traceback (most recent call last):                                                                       
  File "/var/lib/juju/agents/unit-temporal-admin-0/charm/src/charm.py", line 226, in _setup_db_schemas   
    execute(container, "temporal-sql-tool", *command_args)                                               
  File "/var/lib/juju/agents/unit-temporal-admin-0/charm/src/charm.py", line 288, in execute             
    output, warnings = proc.wait_output()                                                                
  File "/var/lib/juju/agents/unit-temporal-admin-0/charm/venv/lib/python3.10/site-packages/ops/pebble.py"
, line 1833, in wait_output                                                                              
    raise ExecError[AnyStr](self._command, exit_code, out_value, err_value)                              ops.pebble.ExecError: non-zero exit code 1 executing ['temporal-sql-tool', '--plugin', 'postgres', '--end
point', 'postgres-primary.temporal-test.svc.cluster.local', '--port', '5432', '--database', 'temporal-k8s_db', '--user', 'relation_id_20', '--password', 'Yg9yg5xFVMFZ5jEs', 'setup-schema', '-v', '0.0'], stdout=
'', stderr='2026-03-23T22:25:51.705Z\t\x1b[31mERROR\x1b[0m\tUnable to connect to SQL database.\t{"error": "pq: remaining connection slots are reserved for non-replication superuser connections", "logging-call-a
t": "handler.go:52"}\n'                                                                                                                                      
The above exception was the direct cause of the following exception:
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nikita-jare**
## Description

This PR addresses flaky terraform apply behaviour in CI / small test models (parallel Juju integrations stressing PostgreSQL and occasional “applications were not available” style errors)
- Explicit `depends_on` on the two charm `module.*` sides for each `juju_integration`, as in [#18](https://github.com/canonical/charmed-temporal-solutions/issues/18)
- `modules/charmed-temporal/test/terraform_test.tfvars` sets `postgresql.config.profile = "testing"`. `validate_test_tfvars` copies that file to the output path and appends model_uuid from the temporal-test Juju model

Closes #19 
Closes #18 
---
## Notes for Reviewers (optional)

---
## Checklist (all that apply)
- [ ] Unit tests added or updated
- [ ] Integration tests added or updated
- [X] Documentation updated

### CODE REVIEW

File: modules/charmed-temporal/justfile [hunk 1]
```diff
@@ -29,6 +29,15 @@ validate_test_tfvars variables_file:
         exit 1
     fi
     echo "model_uuid = \"${MODEL_UUID}\"" > "${variables_file}"
+    {
```
Comment (jananisenthilkumar291) [inline modules/charmed-temporal/justfile]: Consider adding a `test/test.tfvars` files with the config. [Eg](https://github.com/canonical/charmed-airflow-solutions/blob/track/3.1/modules/charmed-airflow/test/terraform_test.tfvars), and append `model_uuid` to the file. Refer [here](https://github.com/canonical/charmed-airflow-solutions/blob/track/3.1/modules/charmed-airflow/justfile).

File: modules/charmed-temporal/integrations.tf [hunk 2]
```diff
@@ -22,33 +26,13 @@ resource "juju_integration" "frontend_visibility_to_postgresql" {
     name     = var.postgresql.app_name
     endpoint = module.postgresql.provides.database
   }
+  depends_on = [
+    module.temporal_frontend,
+    module.postgresql,
+    juju_integration.frontend_to_postgresql,
```
Comment (jananisenthilkumar291) [inline modules/charmed-temporal/integrations.tf]: Looks like `juju_integration.frontend_to_postgresql` already depends on `module.temporal_frontend, module.postgresql`, refer [here](https://github.com/canonical/charmed-temporal-solutions/pull/20/changes#diff-2e8fa32391da559fc402b98c2bf310be2f4c8ca4563abffccec4f2b1b50267efR15)

```suggestion
    juju_integration.frontend_to_postgresql,
```

**Review by shayancanonical — COMMENTED** [review]
Not convinced that we should add `depends_on` chain for production deployments

Postgres connection spikes in production should be able to handle spikes (the spikes resulting from relations being established should be minimal compared to any sustained query traffic from prod systems)

Would you be able to further share motivations (or observed issues)?

I like the use of `profile=testing` in tests though

File: modules/charmed-temporal/integrations.tf [hunk 3]
```diff
@@ -60,6 +44,11 @@ resource "juju_integration" "history_to_postgresql" {
     name     = var.postgresql.app_name
     endpoint = module.postgresql.provides.database
   }
+  depends_on = [
+    module.temporal_history,
+    module.postgresql,
+    juju_integration.frontend_visibility_to_postgresql,
```
Comment (jananisenthilkumar291) [inline modules/charmed-temporal/integrations.tf]: `frontend_visibility_to_postgresql` depends on `frontend_to_postgresql` that depends on `module.postgresql`
```suggestion
    module.temporal_history,
    juju_integration.frontend_visibility_to_postgresql,
```

**Review by jananisenthilkumar291 — COMMENTED** [review]
Added tiny nits. Looks good otherwise. Thanks @nikita-jare.

Agreed with @shayancanonical, that we would want to avoid 'over-engineering' the production Terraform graph with long `depends_on` chains. The proposed set up also does increase the deployment time by having to 'serialize' integrations. 
In order to avoid this, could we consider serializing 'visibility' integrations and/or potentially bump `experimental_max_connections` (refer [here](https://charmhub.io/postgresql-k8s/configurations#:~:text=0%20to%202147483647.-,experimental_max_connections,-%7C%20int)) via `.tfvars`.
In a production env however, I suppose, PostgreSQL can easily fork enough processes to handle multiple concurrent relation joins.

**Comment by nikita-jare** [comment]
Thanks @shayancanonical @jananisenthilkumar291.
I agree on not serializing integrations for every deploy.
We were trying to fix unstable terraform apply in CI and dense test models; many integrations were created in parallel, which led to PostgreSQL connection pressure and occasional “applications were not available to be integrated” style failures. Thought chaining integrations was a straightforward way to tackle that parallelism. I now agree that forcing that long chain on every deployment is a bad default for production applies. I'll remove integration-to-integration chaining and keep explicit `depends_on` on the two charm modules for each relation, as in #18

File: modules/charmed-temporal/integrations.tf (hunk 2 above)
Comment (nikita-jare) [reply modules/charmed-temporal/integrations.tf]: Now with chaining removed, each integration only lists the two modules for that relation

File: modules/charmed-temporal/integrations.tf (hunk 3 above)
Comment (nikita-jare) [reply modules/charmed-temporal/integrations.tf]: Same as above. chaining removed in the latest commit.

File: modules/charmed-temporal/justfile (hunk 1 above)
Comment (nikita-jare) [reply modules/charmed-temporal/justfile]: Done in the latest [commit](https://github.com/canonical/charmed-temporal-solutions/pull/20/commits/4b0fe7789b768908f0abb5f80e9e29c542db7883)

**Comment by nikita-jare** [comment]
Teardown was failing when many juju_integration destroys ran in parallel (provider max duration exceeded) after removing the chaining in the previous [commit](https://github.com/canonical/charmed-temporal-solutions/pull/20/commits/4b0fe7789b768908f0abb5f80e9e29c542db7883)
just destroy now uses terraform destroy -parallelism=1

File: modules/charmed-temporal/justfile [hunk 4]
```diff
@@ -28,7 +28,8 @@ validate_test_tfvars variables_file:
     if ! juju show-model "${MODEL_UUID}" >/dev/null 2>&1; then
         exit 1
     fi
-    echo "model_uuid = \"${MODEL_UUID}\"" > "${variables_file}"
+    cp test/terraform_test.tfvars "${variables_file}"
+    printf '\nmodel_uuid = "%s"\n' "${MODEL_UUID}" >> "${variables_file}"
```
Comment (DnPlas) [inline modules/charmed-temporal/justfile]: (nit, non-blocking) 

```suggestion
    echo '\nmodel_uuid = "%s"\n' "${MODEL_UUID}" >> "${variables_file}"
```

File: modules/charmed-temporal/justfile [hunk 5]
```diff
@@ -28,7 +28,8 @@ validate_test_tfvars variables_file:
     if ! juju show-model "${MODEL_UUID}" >/dev/null 2>&1; then
         exit 1
     fi
-    echo "model_uuid = \"${MODEL_UUID}\"" > "${variables_file}"
+    cp test/terraform_test.tfvars "${variables_file}"
```
Comment (DnPlas) [inline modules/charmed-temporal/justfile]: (nit) I think you also want to remove the variables file to guarantee it does not exist the next time any of these just targets run again from the same directory

**Review by DnPlas — COMMENTED** [review]
Thanks @nikita-jare ! Jut two tiny comments, other than that LGTM. I tested this by deploying the charms with the `test` target and everything worked.

File: modules/charmed-temporal/justfile (hunk 5 above)
Comment (jananisenthilkumar291) [reply modules/charmed-temporal/justfile]: Instead, how would you like if `./test/terraform_test.tfvars` is edited in-line to append `model_uuid` and remove the line on model `destroy` from [Here](https://github.com/canonical/charmed-temporal-solutions/pull/20/changes#diff-0a6f18e8872f23c982edef811195a3a6eadab477480b8180cb8f3aa756416df9L50)

File: modules/charmed-temporal/justfile
Comment (jananisenthilkumar291) [inline modules/charmed-temporal/justfile]:
```suggestion
    just validate_test_tfvars "./test/terraform_test.tfvars"
```

Comment (jananisenthilkumar291) [inline modules/charmed-temporal/justfile]:
```suggestion
    trap 'just destroy "./test/terraform_test.tfvars"' EXIT
```

Comment (jananisenthilkumar291) [inline modules/charmed-temporal/justfile]:
```suggestion
    just apply "./test/terraform_test.tfvars"
```

Comment (jananisenthilkumar291) [inline modules/charmed-temporal/justfile]: I'd suggest to update the `wait-for` for CI to wait until all applications go active.

**Review by jananisenthilkumar291 — CHANGES_REQUESTED** [review]
I still see error on `just apply` and `temporal-admin` is getting stuck in `error` state.
```
    raise ExecError[AnyStr](self._command, exit_code, out_value, err_value)
ops.pebble.ExecError: non-zero exit code 1 executing ['temporal-sql-tool', '--plugin', 'postgres', '--endpoint', 
'postgres-primary.temporal-test.svc.cluster.local', '--port', '5432', '--database', 'temporal-k8s_db', '--user', 'relation_id_19', '--password', 'JW69M7JDPvWIZAXB', 'setup-schema', '-v', '0.0'],
 stdout='', stderr='2026-04-03T18:43:30.684Z\t\x1b[31mERROR\x1b[0m\t**Unable to connect to SQL database.**\t
{"error": "pq: remaining connection slots are reserved for non-replication superuser connections", "logging-call-at": "handler.go:52"}\n'
```

I'd wait till this is resolved. Perhaps you could catch it with a the suggested `juju wait-for`.

File: modules/charmed-temporal/justfile (hunk 5 above)
Comment (DnPlas) [reply modules/charmed-temporal/justfile]: I think that's fine as well

**Comment by nikita-jare** [comment]
Opened `temporal-admin-k8s-operator` PR https://github.com/canonical/temporal-admin-k8s-operator/pull/47 to fix `admin-relation-changed` hook failures and unstable `temporal-admin` during deploy (especially with PgBouncer and several Temporal apps, each with its own admin relation).

Context
`temporal-admin` only sees DB settings from `temporal-k8s` on admin, not from Postgres/PgBouncer directly. That means more relation churn and more chances to run `temporal-sql-tool` while relation data is still incomplete or while the DB path is briefly overloaded.

Changes in admin charm:
- Check `database_connections` before schema work; waiting if it’s not complete yet.
- On `pebble.ExecError` / `pebble.ChangeError` from `temporal-sql-tool`, defer and waiting so transient failures retry.

Once that PR is merged and admin charm is published, the CI issues in this PR will be fixed.

File: modules/charmed-temporal/test/terraform_test.tfvars
Comment (DnPlas) [inline modules/charmed-temporal/test/terraform_test.tfvars:1]: I remember we also wanted to test with the *_max_conns_* values set to the minimum to avoid the issue in https://github.com/canonical/charmed-temporal-solutions/issues/19, but I do not see that here.

Check the temporal-k8s config -> https://charmhub.io/temporal-k8s/configurations#persistence-max-conns

File: modules/charmed-temporal/justfile [hunk 6]
```diff
@@ -53,11 +53,30 @@ destroy variables_file:
     if [ ! -f "${variables_file}" ]; then
         exit 1
     fi
-    terraform destroy -auto-approve -var-file="${variables_file}" || true
+    terraform destroy -auto-approve -parallelism=2 -var-file="${variables_file}" || true
```
Comment (DnPlas) [inline modules/charmed-temporal/justfile]: question: why are we adding parallelism=2 here?

**Review by DnPlas — COMMENTED** [review]
Thanks @nikita-jare, left a couple comments, I think there are some others that haven't been addressed, please check those an make sure they resolved by the reviewers.

File: modules/charmed-temporal/justfile
Comment (nikita-jare) [reply modules/charmed-temporal/justfile]: Addressed here: https://github.com/canonical/charmed-temporal-solutions/pull/20/changes/ab6d9b260aa65dd68a96465b9f2bc3c2ee0bb8f6#diff-0a6f18e8872f23c982edef811195a3a6eadab477480b8180cb8f3aa756416df9R63

Comment (nikita-jare) [reply modules/charmed-temporal/justfile]: Made the change here: https://github.com/canonical/charmed-temporal-solutions/pull/20/changes/ab6d9b260aa65dd68a96465b9f2bc3c2ee0bb8f6#diff-0a6f18e8872f23c982edef811195a3a6eadab477480b8180cb8f3aa756416df9R92

File: modules/charmed-temporal/justfile (hunk 5 above)
Comment (nikita-jare) [reply modules/charmed-temporal/justfile]: Done: https://github.com/canonical/charmed-temporal-solutions/pull/20/changes/ab6d9b260aa65dd68a96465b9f2bc3c2ee0bb8f6#diff-0a6f18e8872f23c982edef811195a3a6eadab477480b8180cb8f3aa756416df9R60

File: modules/charmed-temporal/justfile
Comment (nikita-jare) [reply modules/charmed-temporal/justfile]: Done: https://github.com/canonical/charmed-temporal-solutions/pull/20/changes#diff-0a6f18e8872f23c982edef811195a3a6eadab477480b8180cb8f3aa756416df9R109

Comment (nikita-jare) [reply modules/charmed-temporal/justfile]: Changed here: https://github.com/canonical/charmed-temporal-solutions/pull/20/changes#diff-0a6f18e8872f23c982edef811195a3a6eadab477480b8180cb8f3aa756416df9R118

File: modules/charmed-temporal/justfile (hunk 6 above)
Comment (nikita-jare) [reply modules/charmed-temporal/justfile]: removed here: https://github.com/canonical/charmed-temporal-solutions/pull/20/commits/e158565825102d6d943c724bc873e3f9b89095fb

Had added it earlier as it was failing

File: modules/charmed-temporal/test/terraform_test.tfvars
Comment (nikita-jare) [reply modules/charmed-temporal/test/terraform_test.tfvars:1]: Added theconfig options here: https://github.com/canonical/charmed-temporal-solutions/pull/20/commits/948d3a1f382edd424e706d7dc8eb6678dd6f910a

File: modules/charmed-temporal/justfile (hunk 4 above)
Comment (nikita-jare) [reply modules/charmed-temporal/justfile]: Done here: https://github.com/canonical/charmed-temporal-solutions/pull/20/commits/e158565825102d6d943c724bc873e3f9b89095fb

**Review by DnPlas — DISMISSED** [review]
Thanks @nikita-jare !

File: modules/charmed-temporal/integrations.tf [hunk 7]
```diff
@@ -1,156 +1,214 @@
...
+    endpoint = module.temporal_frontend.requires.ui
+  }
+  application {
+    name     = var.temporal_ui.app_name
+    endpoint = module.temporal_ui.provides.ui
+  }
+  depends_on = [module.temporal_frontend, module.temporal_ui]
+}
+
+# Temporal Frontent ↔ UI (temporal-host-info)
```
Comment (jananisenthilkumar291) [inline modules/charmed-temporal/integrations.tf]: nit
```suggestion
# Temporal Frontend ↔ UI (temporal-host-info)
```

File: modules/charmed-temporal/README.md [hunk 8]
```diff
@@ -32,24 +33,25 @@ The solution module exposes the following configurable inputs:
...
-| `num-history-shards` | string (Temporal only) | Defines number of history shards for the Temporal server | `"1"`                                                         |
+| Field                | Type                   | Description                                              | Default                                                                                     |
+| -------------------- | ---------------------- | -------------------------------------------------------- | ------------------------------------------------------------------------------------------- |
+| `app_name`           | string                 | Application name to deploy                               | Charm-specific                                                                              |
+| `channel`            | string                 | Charm channel to deploy from                             | `"1.23/edge"` for Temporal charms, `"14/stable"` for PostgreSQL, `"1/stable"` for PgBouncer |
+| `revision`           | number                 | Charm revision to pin. `null` means latest for channel.  | `null`                                                                                      |
```
Comment (jananisenthilkumar291) [inline modules/charmed-temporal/README.md]: This is inconsistent with revision defaults set (to 0) in `variables.tf`

File: modules/charmed-temporal/integrations.tf [hunk 9]
```diff
@@ -1,156 +1,214 @@
...
+resource "juju_integration" "frontend_to_ui_host_info"{
+  model_uuid = var.model_uuid
+  application {
+    name     = "temporal-frontend"
+    endpoint = "temporal-host-info"
```
Comment (jananisenthilkumar291) [inline modules/charmed-temporal/integrations.tf]: `temporal-host-info` endpoint is exposed by temporal-ui-k8s-operator. The endpoint value should come from temporal-frontend's `module.requires` similar to [L135](https://github.com/canonical/charmed-temporal-solutions/blob/fix/integration-depends-on/modules/charmed-temporal/integrations.tf#L135).

If you decide to make this change, updates should be made in all temporal repos where this interface is not included as part of their terraform module `output`.

File: modules/charmed-temporal/justfile [hunk 10]
```diff
@@ -82,33 +102,40 @@ goss-cos:
...
 
-    just validate_test_tfvars "terraform_test.tfvars"
+    rm -f terraform.tfstate terraform.tfstate.backup
 
-    just apply-with-overlay "terraform_test.tfvars" "test/terraform_cos.tfvars"
+    just validate_test_tfvars "./test/terraform_test.tfvars"
+
+    just apply-with-overlay "./test/terraform_test.tfvars" "test/terraform_cos.tfvars"
 
     juju wait-for model temporal-test --timeout=600s || true
```
Comment (jananisenthilkumar291) [inline modules/charmed-temporal/justfile]: Can we use the `wait-for-active` target here similar to `test`.

File: modules/charmed-temporal/justfile [hunk 11]
```diff
@@ -121,16 +148,18 @@ test-cos-existing:
...
+
+    just validate_test_tfvars "./test/terraform_test.tfvars"
 
     juju deploy opentelemetry-collector-k8s otel-collector --channel=2/stable --trust -m temporal-test
     juju wait-for application otel-collector --model temporal-test --timeout=300s || true
 
-    just apply-with-overlay "terraform_test.tfvars" "test/terraform_cos_existing.tfvars"
+    just apply-with-overlay "./test/terraform_test.tfvars" "test/terraform_cos_existing.tfvars"
 
     juju wait-for model temporal-test --timeout=600s || true
```
Comment (jananisenthilkumar291) [inline modules/charmed-temporal/justfile]: Can we use the `wait-for-active` target here similar to `test`.

**Review by jananisenthilkumar291 — COMMENTED** [review]
Thanks @nikita-jare, Added a few comment. Happy to approve once these are addressed/considered.

File: modules/charmed-temporal/integrations.tf (hunk 9 above)
Comment (nikita-jare) [reply modules/charmed-temporal/integrations.tf]: Created PRs in the following repos to address this:
https://github.com/canonical/temporal-k8s-operator/pull/137
https://github.com/canonical/temporal-ui-k8s-operator/pull/44
https://github.com/canonical/temporal-admin-k8s-operator/pull/48
https://github.com/canonical/temporal-worker-k8s-operator/pull/92

Fixed in this commit for this repo: https://github.com/canonical/charmed-temporal-solutions/pull/20/commits/d0dc1498d99ef66bbca7785f3cfd54fd1074b6b2

CI will pass once the above PRs are merged

File: modules/charmed-temporal/justfile (hunk 10 above)
Comment (nikita-jare) [reply modules/charmed-temporal/justfile]: Done: https://github.com/canonical/charmed-temporal-solutions/pull/20/commits/d0dc1498d99ef66bbca7785f3cfd54fd1074b6b2

File: modules/charmed-temporal/justfile (hunk 11 above)
Comment (nikita-jare) [reply modules/charmed-temporal/justfile]: Done https://github.com/canonical/charmed-temporal-solutions/pull/20/commits/d0dc1498d99ef66bbca7785f3cfd54fd1074b6b2

File: modules/charmed-temporal/README.md (hunk 8 above)
Comment (nikita-jare) [reply modules/charmed-temporal/README.md]: Addressed in this commit: https://github.com/canonical/charmed-temporal-solutions/pull/20/commits/d0dc1498d99ef66bbca7785f3cfd54fd1074b6b2

File: modules/charmed-temporal/integrations.tf (hunk 7 above)
Comment (nikita-jare) [reply modules/charmed-temporal/integrations.tf]: Changed in this commit: https://github.com/canonical/charmed-temporal-solutions/pull/20/commits/d0dc1498d99ef66bbca7785f3cfd54fd1074b6b2

**Review by jananisenthilkumar291 — DISMISSED** [review]
LGTM! Thanks @nikita-jare

**Review by DnPlas — DISMISSED** [review]

**Review by DnPlas — APPROVED** [review]

**Review by jananisenthilkumar291 — APPROVED** [review]
